data_analysis/correction_framework.py

```diff
@@ -0,0 +1,290 @@
+        if field_to_update == "VOLT":
+            for rec in self.min_records:
+                rec.voltage = funct(rec.voltage)
+        else:
+            print("error: field not recognized/field should not be updated")
+
+    # export night record to a new file
+    def rec_export(self, filename: str):
+        with open(filename, "x") as new_file:
+            if not self.valid_for_use:
+                new_file.write("INVALID\n" + self.exclusion_reason)
+            else:
+                vals = [
+                    self.night_date.strftime(PROJECT_DATE_FORMAT),
+                    self.morning_date.strftime(PROJECT_DATE_FORMAT),
+                    self.sunrise.strftime(PROJECT_DATE_FORMAT),
+                    self.sunset.strftime(PROJECT_DATE_FORMAT),
+                    self.astronomical_twilight.strftime(PROJECT_DATE_FORMAT),
+                    str(self.moon_illumination),
+                ]
+                new_file.write("\n".join(vals) + "\n")
+
+                for rec in self.min_records:
+                    vals = [
+                        rec.utc_time.strftime(PROJECT_DATE_FORMAT),
+                        rec.local_time.strftime(PROJECT_DATE_FORMAT),
+                        str(rec.temperature),
+                        str(rec.frequency),
+                        str(rec.voltage),
+                        rec.sensor_id,
+                    ]
+                    new_file.write(";".join(vals) + "\n")
+
+    # import previously exported night record
+    def rec_import(self, filename: str):
+        # takes an already created object and updates it to imported values
+        with open(filename) as file:
+            p_str = file.readline()
+            if p_str == "INVALID":
+                self.valid_for_use = False
+                self.exclusion_reason = file.readline()
+            else:
+                # night date -- 2023-01-23 06:00:00.992
+                self.night_date = Time(p_str)
+                # morning date
+                self.morning_date = Time(file.readline())
+                # sunrise
+                self.sunrise = Time(file.readline())
+                # sunset
+                self.sunset = Time(file.readline())
+                # twilight
+                self.astronomical_twilight = Time(file.readline())
+                # illumination
+                self.moon_illumination = float(file.readline())
+
+                p_str = file.readline()
+                while len(p_str) != 0:
+                    data = p_str.split(";")
+                    m_rec = MinuteRecord(
+                        Time(data[0]),
+                        Time(data[1]),
+                        float(data[2]),
+                        float(data[3]),
+                        float(data[4]),
+                        data[5],
+                    )
+                    self.min_records.append(m_rec)
+                    p_str = file.readline()
+
+
+def cor_framework_tests():
```

Comment:
You can run tests using `pytest`.  An example of Pythonic test functions would be like this: https://github.com/chanteriam/reproductive-rights-data-project/blob/fa6250974ec4fd604b5b1d0e9ee9e65f2f4d9cf1/reproductive_rights_project/test/test_util.py

Reply:
I see! I wasn't sure how to execute functions in Python, much less tests, so I went with prints partially for that reason and partially for presentation reasons if it needed to be seen in action.